python/tk_desktop/desktop_window.py

```diff
@@ -40,6 +42,7 @@
 from .project_commands_model import ProjectCommandModel
 from .project_commands_model import ProjectCommandProxyModel
 from .project_commands_widget import ProjectCommandDelegate
+from shotgun_desktop.desktop_login import get_shotgun_authenticator
```

Comment:
Good point! 
The reason there is a dependency is that desktop-internal has its own defaults manager which handles some of the desktop specific logic for how default values etc are saved. This is different in the desktop vs core - the desktop using the login fw/keychain vs core picking up some stuff from shotgun.yml. Eventually, if desktop and core converges on this functionality, the external module will take care of everything. Right now, since there are differences, core and desktop-internal each implements a small class to define certain behaviour around storing of defaults etc. 

But the code executed right here is just handling logging out. I wonder if the code could simply be

```
from tank_vendor.shotgun_authentication import ShotgunAuthenticator
shotgun_auth = ShotgunAuthenticator()
shotgun_auth.clear_saved_user()
```